feat(ephemeris): add convert ICRS to beam-model XY

rikvl opened the pull request:
Add function to convert ICRS equatorial coordinates to CHIME/FRB beam-model XY coordinates

Should do the same as the CHIME/FRB beam-model utility `get_position_from_equatorial`:
https://github.com/CHIMEFRB/beam-model/blob/068e410a99bfb18e469a878e659c26393278874f/beam_model/utils.py#L122

rikvl commented:
@jrs65 , @arashcosmology This PR can be reviewed.

The PR now also contains helper functions to convert between beam-model XY and CIRS hour angle and declination (in both directions). I could still add tests.

Manually I did the following tests:
- Round trip.
- Difference in Ha and Dec between these helper functions and equivalent parts of functions in the `beam_model` package.

The latter passes when going from HA, Dec to beam-model X, Y: all differences < 10^-10.
However, when going from beam-model X, Y to HA, Dec, there are big remaining differences for inputs below the horizon (see graph). I added a warning for inputs below the horizon, and I'm inclined to let this be -- but let me know what you think.

![diff-ha-dec](https://github.com/chime-experiment/ch_util/assets/52370634/5e02c7fb-d9dd-4104-a086-436db31ffcfb)

ketiltrout reviewed (changes requested):
This looks fairly straightforward, but you have added a dependency on `driftscan` which needs to be added to the requirements file.

You've included a `bmxy_to_hadec`, but is there no need for a `bmxy_to_equat`?

rikvl commented on ch_util/ephemeris.py:
```diff
@@ -696,6 +696,170 @@ def object_coords(body, date=None, deg=False, obs=chime):
+
+    from ch_util.tools import _CHIME_ROT
+
+    from drift.telescope.cylbeam import rotate_ypr
```
Added.

rikvl commented on ch_util/ephemeris.py:
```diff
@@ -696,6 +696,170 @@ def object_coords(body, date=None, deg=False, obs=chime):
+        The ICRS Right Ascension of the source in degrees.
+    dec_icrs : float
+        The ICRS Declination of the source in degrees.
+    time : array_like
```
Removed the entire `equat_to_bmxy` function.

rikvl commented:
Thanks Don!

I added `driftscan` to the dependencies.

Regarding the different functions:
- `bmxy_to_hadec` is the only one really required, I believe.
- `hadec_to_bmxy` was added for completeness.
- `equat_to_bmxy` was a remnant from earlier stages, and not required. Given the issue with the `time` argument that you spotted (which is important), I removed it for now.

ketiltrout reviewed (approved):
Looks good.
